src/main/java/edu/harvard/iq/dataverse/DatasetField.java

```diff
@@ -464,11 +466,50 @@ public boolean isRequired() {
             {
                 required = false;
             }
+            
```

Comment:
are lines 465-468 needed with the new code below? Seems like if a compound field is required for some reason - required is true at line ~462 but would be set false if there are no required children, and then explicitly set true of there are, so the initial value at line 462 doesn't seem to be used and the next two ifs seem to do almost the same thing (would be exact except isHasRequiredChildrenDV is slightly different than isHasRequiredChildren).

Reply:
465 - 468 (which were put in back in 2021) are there for an erroneously set up metadata block where the parent is set to required, but none of the children are required. Maybe we could or should do something when the metadata block is added or edited, but again the code below is to maintain the conditionally required children by only setting the parent to required if the child is set to required at the DV level.